release.sh

```diff
@@ -16,69 +16,188 @@ BLUE='\033[0;34m'
+  exit 1
+fi
+
+# ---
+# Check environment variables
+# ---
+
+check_env_var() {
+  local var_name=$1
+  if [ -z "${!var_name}" ]; then
+    echo -e "${RED}⚠️  La variable d'environnement $var_name doit être définie.${NC}"
+    exit 1
+  fi
+}
 
-  echo "Changement de branche vers release..."
-  git switch release
+echo -e "Vérification des variables d'environnement"
+check_env_var "SCALINGO_REGION"
+check_env_var "SCALINGO_BACK_APP"
+check_env_var "SCALINGO_FRONT_APP"
 
-  # Vérifier s'il y a des commits d'écart entre 'main' et 'release'
-  COMMITS_DIFF=$(git rev-list --count release..main)
+# ---
+# Check Scalingo concerns
+# ---
 
-  if [ "$COMMITS_DIFF" -gt 0 ]; then
-    echo "Il y a $COMMITS_DIFF commit(s) d'écart entre 'main' et 'release' pour $repo_name. Lancement du merge..."
-    
-    echo "Merge de la branche main dans release pour $repo_name..."
-    if ! git merge --no-ff main -m "$MERGE_COMMIT_MESSAGE"; then
-      echo "⚠️ Conflit lors du merge de la branche main dans release pour $repo_name. Veuillez résoudre manuellement."
-      exit 1
-    fi
+echo -e "Vérification du CLI Scalingo"
+if ! command -v scalingo &> /dev/null; then
+  echo -e "${RED}⚠️  Le CLI Scalingo n'est pas installé. Veuillez l'installer avant d'exécuter ce script.${NC}"
+  exit 1
+fi
 
-    echo "Poussée des changements vers le dépôt distant pour $repo_name..."
-    git push
+echo -e "Vérification des accès aux applications Scalingo"
+APPS_LIST=$(scalingo apps --region "$SCALINGO_REGION")
 
-    echo "✅ La branche 'release' du repository $repo_name a été mise à jour"
+check_app_access() {
+  local app_name=$1
+  if echo "$APPS_LIST" | grep -E "^\|[[:space:]]*$app_name[[:space:]]*\|[[:space:]]*collaborator[[:space:]]*\|" &> /dev/null; then
+    echo -e "Accès confirmé pour l'application '$app_name'."
   else
-    echo "🙅 Pas de commits à merger entre 'main' et 'release' pour $repo_name. Aucun merge effectué."
+    echo -e "${RED}⚠️  Vous n'avez pas accès en tant que collaborateur à l'application '$app_name' dans la région '$SCALINGO_REGION'. Veuillez vérifier vos permissions.${NC}"
+    exit 1
   fi
+}
 
-  # Revenir au répertoire temporaire
-  cd ..
-  echo ""
+check_app_access "$SCALINGO_BACK_APP"
+check_app_access "$SCALINGO_FRONT_APP"
+
+# ---
+# Utils
+# ---
+
+get_next_version() {
+  local version=$1
+  local release_type=$2
+
+  # Supprimer le 'v' au début et séparer en parties (major.minor.patch)
+  version="${version#v}"
+  IFS='.' read -r major minor patch <<< "$version"
+
+  case $release_type in
+    major)
+      major=$((major + 1))
+      minor=0
+      patch=0
+      ;;
+    minor)
+      minor=$((minor + 1))
+      patch=0
+      ;;
+    patch)
+      patch=$((patch + 1))
+      ;;
+    *)
+      echo -e "${RED}⚠️  Type de release non valide. Utilisez major, minor ou patch.${NC}"
+      exit 1
+      ;;
+  esac
+
+  echo "v$major.$minor.$patch"
```

Comment:
_nit_ :
Faut aimer le shell :)
Sinon, plutot que spliter explicitement le semver, y a des trucs qui le font en bash ou autre.

Reply:
T'as un exemple de bloc de commandes / commande magique simple ?

J'ai trouvé un utilitaire – [semver-tool](https://github.com/fsaintjacques/semver-tool) – qui a l'air de faire le job, mais je ne suis pas fan de devoir installer une dépendance.

Sur StackOverflow, je trouve les solutions proposées – [dont ce thread](https://stackoverflow.com/questions/6245293/extract-version-number-from-file-in-shell-script) – pas totalement simples à comprendre / maintenir.

Reply:
Discuté en grand séminaire. On peut envoyer en l'état.